docsTech/processing.md

```diff
@@ -0,0 +1,272 @@
+# Processing
+
+## Introduction
+
+## Pipe-processing
+
+A pipe processing is a pattern allowing plugins or components to enrich data used by another plugin.
+
+We can classify the pipe-processing into several categories:
+
+### Plugin state enrichment
+
+**Goal**: Allow plugins to enrich another plugin state before saving it.
+
+#### Processing list
+
+#### `hydrateColumns`
+
+**Publisher**: `useGridColumns` plugin before updating `state.columns`.
+
+**Why register to this processing**: Add some columns (eg: processor of the Selection plugin) or re-order the columns (eg: processor of the Column Pinning plugin).
+
+**Example**:
+
+```ts
+const addCustomFeatureColumn = React.useCallback<GridPipeProcessor<'hydrateColumns'>>(
+  (columnsState) => {
+    const customFeatureColumn = getCustomFeatureColumn();
+    const shouldHaveCustomFeatureColumn = !props.disableCustomFeature;
+    const haveCustomFeatureColumn = columnsState.lookup[customFeatureColumn.field] != null;
+
+    if (shouldHaveCustomFeatureColumn && !haveCustomFeatureColumn) {
+      columnsState.lookup[customFeatureColumn.field] = customFeatureColumn;
+      columnsState.all = [customFeatureColumn.field, ...columnsState.all];
+    } else if (!shouldHaveCustomFeatureColumn && haveCustomFeatureColumn) {
+      delete columnsState.lookup[customFeatureColumn.field];
+      columnsState.all = columnsState.all.filter((field) => field !== customFeatureColumn.field);
+    }
+
+    return columnsState;
+  },
+  [apiRef, classes, getSelectionColumn],
+);
+
+useGridRegisterPipeProcessor(apiRef, 'hydrateColumns', updateSelectionColumn);
+```
+
+> ⚠ The `columnsState` passed to the processors can contain the columns returned by the previous processing.
+> If the plugin is not enabled during the current processing, it must check if its columns are present, and if so remove them.
+
```

Comment:
If this is related to the example. It would be easier to understand as a comment in the code